Fix compilation. Already in 2.6.10 tree.

# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2004/10/18 22:43:26-07:00 torvalds@ppc970.osdl.org 
#   Wrap <linux/compiler.h> inside '#ifndef __ASSEMBLY__'
#   
#   None of the compatibility defines make sense for assembly
#   files, and gcc has trouble with vararg macros when using
#   "-traditional" (which is used for asm), to the point of
#   ICE'ing.
# 
# include/linux/compiler.h
#   2004/10/18 22:43:20-07:00 torvalds@ppc970.osdl.org +4 -2
#   Wrap <linux/compiler.h> inside '#ifndef __ASSEMBLY__'
#   
#   None of the compatibility defines make sense for assembly
#   files, and gcc has trouble with vararg macros when using
#   "-traditional" (which is used for asm), to the point of
#   ICE'ing.
# 
Index: linux-2.6.9-ck1/include/linux/compiler.h
===================================================================
--- linux-2.6.9-ck1.orig/include/linux/compiler.h	2004-10-19 08:57:12.000000000 +1000
+++ linux-2.6.9-ck1/include/linux/compiler.h	2004-10-19 20:05:41.255008517 +1000
@@ -1,6 +1,8 @@
 #ifndef __LINUX_COMPILER_H
 #define __LINUX_COMPILER_H
 
+#ifndef __ASSEMBLY__
+
 #ifdef __CHECKER__
 # define __user		__attribute__((noderef, address_space(1)))
 # define __kernel	/* default address space */
@@ -21,7 +23,6 @@ extern void __chk_io_ptr(void __iomem *)
 
 #ifdef __KERNEL__
 
-#ifndef __ASSEMBLY__
 #if __GNUC__ > 3
 # include <linux/compiler-gcc+.h>	/* catch-all for GCC 4, 5, etc. */
 #elif __GNUC__ == 3
@@ -31,7 +32,6 @@ extern void __chk_io_ptr(void __iomem *)
 #else
 # error Sorry, your compiler is too old/not recognized.
 #endif
-#endif
 
 /* Intel compiler defines __GNUC__. So we will overwrite implementations
  * coming from above header files here
@@ -61,6 +61,8 @@ extern void __chk_io_ptr(void __iomem *)
     (typeof(ptr)) (__ptr + (off)); })
 #endif
 
+#endif /* __ASSEMBLY__ */
+
 #endif /* __KERNEL__ */
 
 /*
